### DISCUSSION TITLE
How can we remove the options from last execution when rerun failed test?

### DISCUSSION THREAD

**Discussion by windyliu1118** [getgauge/gauge#1818] [General]
When I use the rerun failed command `gauge run --failed`, there will be some concerns from my side:
1. When there is no failure, then when we execute the rerun failed command in jenkins job, the build status will be marked as failed.
2. When I run tests in parallel at the beginning, and then rerun the failed test, then the related specs with failed tests are executed again in parallel.
So I want to ask:
1. If there is no failure, when run the rerun failed command, can we get the jenkins job as Pass?
2. Can we rerun the failed test without the execution options in last run?
~~@zabil can you take a look at my questions? Thanks a lot.~~

**Comment by zabil** [comment]
> @zabil can you take a look at my questions? Thanks a lot.

It's best to keep questions open so that everyone can participate. In order to do that it's best not to call out specific users like above. That said I will answer this question, but in the future I will edit the original message to remove sections calling out specific users.

> When there is no failure, then when we execute the rerun failed command in jenkins job, the build status will be marked as failed.

The status code will always be a failure status code for failed tests re-rerun. 

> If there is no failure, when run the rerun failed command, can we get the jenkins job as Pass?

I am not sure about Jenkins. But you will have to ignore the exit status code. You can try searching for "Ignore failure status code in jenkins"

> Can we rerun the failed test without the execution options in last run?

I don't think so. This is also mentioned here https://docs.gauge.org/execution.html?os=macos&language=javascript&ide=vscode#rerun-failed-scenarios
